charts/ark-solr/templates/statefulset.yaml

```diff
@@ -0,0 +1,63 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "solr.fullname" . }}-statefulset
+  labels:
+    {{- include "solr-helm-charts.labels" . | nindent 6 }}
+  namespace: {{ .Release.Namespace | quote }}
+spec:
+  replicas: 1
+  selector: 
+    matchLabels:  
+      {{- include "solr-helm-charts.labels" . | nindent 6 }}
+  serviceName: {{ include "solr.fullname" . }}-headless
+  template:
+    metadata:
+      name: ark-solr
+      labels:
+        {{- include "solr-helm-charts.labels" . | nindent 8 }}
+    spec:
+      initContainers:
+      containers:
+      - name: solr
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        ports:
+        - containerPort: 8983
+          name: solr-port
+          protocol: TCP
+        command:
+        livenessProbe:
+        readinessProbe: 
+        resources:
+        volumeMounts:
+        - name: solr-data
+          mountPath: /data
+        - name: solr-conf
+          mountPath: /opt/solr/server/solr/configsets/_default/conf
+      volumes:
+        - name: solr-conf
+          configMap: 
+            name: {{ include "solr.configmap-name" . }}
```

Comment:
@naveen-armedia @JonHolman @drivera-armedia We need to have a discussion on the configuration of containerized apps. As I mentioned during my presentation of the deployment script, I would be in favour of using environment variables, which would mean that the Docker images have configuration files (or configuration file templates), along with a startup script to update those confguration template files.